### PR TITLE
Typing fixes

### PR DESCRIPTION
**Pull request by max-radin**
This fixes mypy errors in a few different files.

### CODE REVIEW

File: src/python/zquantum/core/bitstring_distribution/distance_measures/mmd.py [hunk 1]
```diff
@@ -62,14 +62,13 @@ def compute_multi_rbf_kernel(x_i: np.array, y_j: np.array, sigmas: List) -> np.n
...
     exponent = np.abs(x_i[:, None] - y_j[None, :]) ** 2
-    kernel_matrix = 0.0
     for sigma in sigmas:
         try:
             gamma = 1.0 / (2 * sigma)
         except ZeroDivisionError as error:
             print("Handling run-time error:", error)
             raise
-        kernel_matrix = kernel_matrix + np.exp(-gamma * exponent)
+        kernel_matrix = np.exp(-gamma * exponent)
```
Comment (mstechly) [inline src/python/zquantum/core/bitstring_distribution/distance_measures/mmd.py:71]: I think you have changed the logic here. Shouldn't it be `+=`?

File: src/python/zquantum/core/measurement.py [hunk 2]
```diff
@@ -659,7 +674,11 @@ def concatenate_expectation_values(
         The combined expectation values.
     """
 
-    combined_expectation_values = ExpectationValues(np.zeros(0,))
+    combined_expectation_values = ExpectationValues(
```
Comment (mstechly) [inline src/python/zquantum/core/measurement.py:677]: Actually, it'll be cleaner if you just remove `,` after 0 and blackify it :)

**Review by mstechly — CHANGES_REQUESTED** [review]

File: src/python/zquantum/core/bitstring_distribution/distance_measures/mmd.py (hunk 1 above)
Comment (max-radin) [reply src/python/zquantum/core/bitstring_distribution/distance_measures/mmd.py:71]: Ah you're right. I had seen that `kernel_matrix` was being initialized to zero, but didn't see that the addition was happening in a loop.

Comment (alexjuda) [reply src/python/zquantum/core/bitstring_distribution/distance_measures/mmd.py]: fixed in 2d5abfa

File: src/python/zquantum/core/measurement.py (hunk 2 above)
Comment (alexjuda) [reply src/python/zquantum/core/measurement.py]: Done in d8682df...2c47b89

File: src/python/zquantum/core/bitstring_distribution/distance_measures/mmd.py (hunk 1 above)
Comment (mstechly) [reply src/python/zquantum/core/bitstring_distribution/distance_measures/mmd.py]: Power of PRs!

**Review by mstechly — APPROVED** [review]

**Comment by mstechly** [comment]
Ready to go after conflicts are resolved.

**Pull request merged**